### PR TITLE
fix: added base url for mailgun provider

### PR DESCRIPTION
**Pull request by venarius**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
fixes https://github.com/novuhq/novu/issues/1251

  This adds the base url option for the mailgun provider

### CODE REVIEW

File: providers/mailgun/src/lib/mailgun.provider.spec.ts [hunk 1]
```diff
@@ -20,16 +19,6 @@ const mockNovuMessage = {
 
 test('should trigger mailgun correctly', async () => {
   const provider = new MailgunEmailProvider(mockConfig);
-  const createFn = jest.fn();
```
Comment (scopsy) [inline providers/mailgun/src/lib/mailgun.provider.spec.ts:23]: Good to remove this! Thanks <3

File: providers/mailgun/src/lib/mailgun.provider.spec.ts [hunk 2]
```diff
@@ -40,5 +29,25 @@ test('should trigger mailgun correctly', async () => {
 
   await provider.sendMessage(mockNovuMessage);
 
+  expect(api.isDone()).toBeTruthy();
+  api.done();
+});
+
+test('should trigger mailgun correctly with custom baseUrl', async () => {
```
Comment (scopsy) [inline providers/mailgun/src/lib/mailgun.provider.spec.ts:36]: Love the test!

File: providers/mailgun/package.json [hunk 3]
```diff
@@ -46,7 +46,7 @@
   "dependencies": {
     "@novu/stateless": "^0.7.2",
     "form-data": "^4.0.0",
-    "mailgun.js": "^3.5.8",
+    "mailgun.js": "^8.0.1",
```
Comment (scopsy) [inline providers/mailgun/package.json:49]: This was our of date :o

**Review by scopsy — APPROVED** [review]
Amazing! Thanks for this 🚀

File: providers/mailgun/package.json (hunk 3 above)
Comment (venarius) [reply providers/mailgun/package.json:49]: A tad 😅

**Pull request merged**